### DISCUSSION TITLE
Is this maintaind?

### DISCUSSION THREAD

**Discussion by Cuteistfox** [libsixel/libsixel#76] [Q&A] [answered]
no update sence 2022

**Comment by dankamongmen** [comment, marked as answer]
i am not maintaining it, and honestly suggest it be archived.

**Reply by krinkleneck** [reply]
Darn, I was hoping to add the original DEC documentation to the mix for anyone who wanted the specification to work from right there.

**Comment by GitMurf** [comment]
I just stumbled upon the notice from 2020 that the original repo had gone unmaintained and so this fork was created... see here: https://github.com/saitoha/libsixel/issues/154

I am not trying to be ironic here, but has this fork seen the same fate? Is it in a state that still should be relied upon over the original repo (https://github.com/saitoha/libsixel)? Or at this point should we be using the original (also unmaintained) repo?

Thanks in advance @dankamongmen and @ctrlcctrlv for your response / feedback!

**Reply by dankamongmen** [reply]
this library ought not be used imho. API decisions made by the original author make it impossible to use securely, and there are better options for new code.

**Reply by GitMurf** [reply]
@dankamongmen thanks for the quick response! By “this library” do you mean both the original repo and this fork?

Could you kindly suggest alternatives that you recommend? I am really just looking for the functionality of img2sixel for rendering images in my terminal. Thanks so much!

**Reply by dankamongmen** [reply]
yes, the original repo and this fork. the original design was a fundamentally flawed design; i didn't realize as much when i agreed to assist in maintaining this fork (essentially, you cannot supply a length parameter, and thus a malformed input can cause the lexer to run off valid memory IIRC).

i recommend my own notcurses, but i'm biased.

**Comment by Kreijstal** [comment]
Why not make a libsixel2.0 with new api convention?

**Reply by Kreijstal** [reply]
Btw I could lazy mantain it, a #mob fork, make test, CI, if it passes, good to go

**Reply by j4james** [reply]
Honestly the reports of the API being insecure are overblown IMO. As far as I recall, there was a bug report from a user doing the equivalent of something like this (I don't remember the exact API):

```c
char buffer[10];
encode(buffer, 800, 600, ...);
```

Where 800 and 600 were the width and height of the image that was supposed to be written to the buffer. Since 800*600 is obviously greater than 10, it overruns the buffer and crashes. But that's not an API problem - that's a user problem.

Could the API be made more user friendly so that an error like that is less likely? Sure. But that doesn't mean it's impossible to write secure code using the current API.

Better documentation would be a good first step. Like on functions that take a buffer, make it clear exactly how big the buffer needs to be, because that's not always obvious (like if there's a bpp parameter, it's more complicated than just width*height).

It's also quite likely there are other bugs in the code that make it insecure, so those would need to be fixed. But I've yet to see evidence of anything that requires the entire API to be scrapped. It just needs someone willing to take on the maintenance burden.

**Reply by dankamongmen** [reply]
go for it

**Reply by dankamongmen** [reply]
> Where 800 and 600 were the width and height of the image that was supposed to be written to the buffer. Since 800*600 is obviously greater than 10, it overruns the buffer and crashes. But that's not an API problem - that's a user problem.
> Could the API be made more user friendly so that an error like that is less likely? Sure. But that doesn't mean it's impossible to write secure code using the current API.

i believe it was stronger than this -- iirc, you couldn't determine the necessary output buffer size, at least. but i am not certain.

**Comment by dankamongmen** [comment]
anyway, if someone wants to take it over, i wholeheartedly encourage it

**Reply by Kreijstal** [reply]
How are you going to choose a mantainer?
